packages/ui-components/src/chat/chat.tsx

```diff
@@ -0,0 +1,228 @@
+/**
+ * Copyright 2023 Impulse Innovations Limited
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as React from 'react';
+import { v4 as uuidv4 } from 'uuid';
+
+import styled from '@darajs/styled-components';
+import { Xmark } from '@darajs/ui-icons';
+
+import Button from '../button/button';
+import TextArea from '../textarea/textarea';
+import { InteractiveComponentProps, Message } from '../types';
+import { default as MessageComponent } from './message';
+
+const ChatWrapper = styled.div`
+    overflow-y: auto;
+    display: flex;
+    flex-direction: column;
+    gap: 1rem;
+
+    width: 350px;
+    height: calc(100vh - 2rem);
+    margin: 1rem;
+    padding: 1.5rem;
+
+    background-color: ${(props) => props.theme.colors.background}e6;
+    border-radius: 0.4rem;
+    box-shadow: ${(props) => props.theme.shadow.medium};
+`;
+
+const ReplyWrapper = styled.div`
+    display: flex;
+    flex-direction: column;
+    gap: 1rem;
+    justify-self: flex-end;
+
+    height: 8.6rem;
+    margin-top: auto;
+`;
+
+const ReplyButtons = styled.div`
+    display: flex;
+    gap: 1rem;
+    justify-content: flex-end;
+`;
+
+const ChatBody = styled.div`
+    overflow-y: auto;
+    display: flex;
+    flex-direction: column;
+    gap: 1rem;
+
+    max-height: calc(100% - 10.225rem);
+    margin: -0.25rem;
+    padding: 0.25rem;
+`;
+
+const ChatTop = styled.div`
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+
+    width: 100%;
+    height: 1.625rem;
+
+    font-size: 1.2rem;
+`;
+
+const CloseIcon = styled(Xmark)`
+    height: 1.2rem;
+    color: ${(props) => props.theme.colors.grey4};
+
+    :hover {
+        color: ${(props) => props.theme.colors.grey5};
+    }
+
+    :active {
+        color: ${(props) => props.theme.colors.grey6};
+    }
+`;
+
+export interface ChatProps extends InteractiveComponentProps<Message[]> {
+    /** Event triggered when a message is added */
+    onAdd?: (value: Message[]) => void | Promise<void>;
+    /** Event triggered when the chat sidebar is closed */
+    onClose?: () => void | Promise<void>;
+    /** Event triggered when a message is deleted */
+    onDelete?: (id: string) => void | Promise<void>;
+    /** Event triggered when a message is edited */
+    onEdit?: (value: Message[]) => void | Promise<void>;
```

Comment:
What's the reason to expose `onAdd`, `onDelete` and `onEdit` rather than just one `onUpdate`?

Reply:
Good point, have updated this